src/components/inputs/text/NameTokenSelector/NameTokenSelector.tsx

```diff
@@ -30,177 +44,512 @@ function NameTokenSelector({
+                .map((token, index) => {
+                  if (!token) {
+                    return;
+                  }
+
+                  return (
+                    <button
+                      key={index}
+                      className="name-token-item pointer"
+                      onClick={() => {
+                        handleSetToken({
+                          id: token?.id,
+                          name: token.name ?? '',
+                          ticker: token.ticker ?? '',
+                        });
+                      }}
+                    >
+                      {token.name && token.ticker
+                        ? `${token.name} (${token.ticker}) - ${token.id}`
+                        : token.id}
+                    </button>
+                  );
+                })
+            ) : (
+              Object.entries(tokens)
+                .slice(
+                  Math.max((listPage - 1) * listItemCount, 0),
+                  listPage * listItemCount,
+                )
+                .map((token, index) => {
+                  if (!token) {
+                    return;
+                  }
+                  const [id, details] = token;
+                  const { name, ticker, names } = details;
+
+                  return (
+                    <button
+                      key={index}
+                      className="name-token-item pointer"
+                      onClick={() => {
+                        handleSetToken({
+                          id,
+                          name: name ?? '',
+                          ticker: ticker ?? '',
+                        });
+                      }}
+                    >
+                      {name && ticker ? `${name} (${ticker}) - ${id}` : id}
+                      {names?.length ? (
+                        <HamburgerOutlineIcon
+                          width={20}
+                          height={20}
+                          fill="var(--text-faded)"
+                        />
+                      ) : (
+                        <></>
+                      )}
+                    </button>
+                  );
+                })
+            )}
+            <div
+              className="custom-next-pagination flex flex-column"
               style={{
-                gap: '1em',
-                border: '1px solid var(--text-faded)',
-                borderRadius: '50px',
-                padding: '3px 5px',
+                padding: '10px 25px',
+                boxSizing: 'border-box',
+                width: '100%',
+                justifyContent: 'flex-start',
               }}
-              onClick={() => setSelectedNameToken(undefined)}
             >
-              Remove
-              <CircleXIcon
-                width={'20px'}
-                height={'20px'}
-                fill={'var(--text-faded)'}
+              <Pagination
+                total={
+                  tokens && !filteredTokens
```

Comment:
Same here: `tokens` is non-nullable by this point

Reply:
fixed in https://github.com/ar-io/pdns-react/pull/229/commits/4de5520d092e8e98eaa9dcffe69150aac929361b